# HG changeset patch
# User Rik <rik@octave.org>
# Date 1544489902 28800
#      Mon Dec 10 16:58:22 2018 -0800
# Node ID b8f23ab958e84620e2aac72556610f17c6892703
# Parent  a5094f65012c31cacdc074e3ed050bfcba6db27d
surfl.m: Fix input validation for 4x4 input matrices (bug #55180)

* surfl.m: Check that any material property input or light source location
input is a vector before presuming it to be one of those.

diff --git a/scripts/plot/draw/surfl.m b/scripts/plot/draw/surfl.m
--- a/scripts/plot/draw/surfl.m
+++ b/scripts/plot/draw/surfl.m
@@ -104,27 +104,28 @@ function h = surfl (varargin)
   endif
 
   ## Check for reflection properties argument.
   ##
   ## r = [ambient light strength,
   ##      diffuse reflection strength,
   ##      specular reflection strength,
   ##      specular shine]
-  if (isnumeric (varargin{end}) && length (varargin{end}) == 4)
+  if (isnumeric (varargin{end}) && isvector (varargin{end})
+      && (numel (varargin{end}) == 4))
     r = varargin{end};
     varargin(end) = [];
   else
     ## Default values.
     r = [0.55, 0.6, 0.4, 10];
   endif
 
   ## Check for light vector (lv) argument.
   have_lv = false;
-  if (isnumeric (varargin{end}))
+  if (isnumeric (varargin{end}) && isvector (varargin{end}))
     len = numel (varargin{end});
     lastarg = varargin{end};
     if (len == 3)
       lv = lastarg;
       varargin(end) = [];
       have_lv = true;
     elseif (len == 2)
       [lv(1), lv(2), lv(3)] = sph2cart ((lastarg(1) - 90) * pi/180,
